ui-components/src/helpers/filters.ts

```diff
@@ -39,21 +42,42 @@ export function encodeToFrontendFilterString(filterParams: ListingFilterParams)
   let queryString = ""
   for (const filterType in filterParams) {
     const value = filterParams[filterType]
-
     if (filterType in ListingFilterKeys && value !== undefined && value !== "") {
       queryString += `&${filterType}=${value}`
     }
   }
   return queryString
 }
 
+export enum AvailabilityFilterType {
+  any = "any",
+  hasAvailability = "hasAvailability",
+  noAvailability = "noAvailability",
+  waitlist = "waitlist",
+}
+
+export type FormFilterData = ListingFilterParams & {
+  availability?: AvailabilityFilterType
+}
+
```

Comment:
Is there a downside to having this enum in the backend and then doing this translation of enum type -> SQL query in the backend instead of the frontend?

Reply:
There is some funkiness with the filter mappings to columns, but otherwise no. I just did it this way because I missed that the backend is already set up to handle custom filters, so I was going out of my way to keep the backend generic. Updated with custom backend filter handling.